pandas/core/array_algos/take.py

```diff
@@ -142,8 +148,11 @@ def take_1d(
         # ExtensionArray -> dispatch to their method
         return arr.take(indexer, fill_value=fill_value, allow_fill=allow_fill)
 
+    if not allow_fill:
+        return arr.take(indexer)
+
     indexer, dtype, fill_value, mask_info = _take_preprocess_indexer_and_fill_value(
-        arr, indexer, 0, None, fill_value, allow_fill
+        arr, indexer, None, fill_value, allow_fill
```

Comment:
do we ever get to _take_preprocess_indexer_and_fill_value with allow_fill=False?

Reply:
Yes, that's currently possible in `take_nd`, and a quick search reveals that we have a few cases throughout pandas that calls `take_nd` specifying this argument. That's probably something to optimize though.